sktime/annotation/base/_base.py

```diff
@@ -0,0 +1,435 @@
+# -*- coding: utf-8 -*-
+"""
+Base class template for annotator base types:
+ - BaseStreamAnnotator - annotating a data stream, with temporal update
+ - BasePanelAnnotator - annotating a panel of sequences or time series
+
+ Class summaries below:
+
+---
+
+    class name: BasePanelAnnotator
+
+Scitype defining methods:
+    fitting              - fit(self, X, Y=None, Z=None)
+    annotating           - predict(self, X, Z=None)
+
+    NOT YET IMPLEMENTED
+    updating (minibatch) - update(self, X, Y=None, Z=None)
+    update&annotate      - update_predict(self, X, X_new, Y=None, Z=None, Z_new=None)
+
+Inspection methods:
+    hyper-parameter inspection  - get_params()
+    fitted parameter inspection - get_fitted_params()
+
+State:
+    fitted model/strategy   - by convention, any attributes ending in "_"
+    fitted state flag       - check_is_fitted()
+
+---
+
+    class name: BaseStreamAnnotator
+
+Scitype defining methods:
+    fitting              - fit(self, X, Y=None)
+    annotating           - predict(self, X)
+    updating (temporal)  - update(self, X, Y=None)
+    update&annotate      - update_predict(self, X)
+
+Inspection methods:
+    hyper-parameter inspection  - get_params()
+    fitted parameter inspection - get_fitted_params()
+
+State:
+    fitted model/strategy   - by convention, any attributes ending in "_"
+    fitted state flag       - check_is_fitted()
+
+copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""
+
+__author__ = ["satya-pattnaik ", "fkiraly"]
+
+from sktime.base import BaseEstimator
+
+from sktime.utils.validation.series import check_series
+
+import pandas as pd
+
+
+class BasePanelAnnotator(BaseEstimator):
+    """Base annotator
+
+    The base annotator specifies the methods and method
+    signatures that all annotators have to implement.
+
+    Specific implementations of these methods is deferred to concrete
+    annotators.
+    """
+
+    # default tags
+    _tags = {
+        "handles-panel": True,  # can handle panel annotations, i.e., list X/y?
+        "handles-missing-data": False,  # can handle missing data in X, Y, Z
+        "annotation-type": "point",  # can be point, segment or both
+        "annotation-labels": "none",  # can be one of, or list-subset of
+        #   "label", "outlier", "change"
+    }
+
+    def __init__(self):
+
+        self._is_fitted = False
+
+        super(BasePanelAnnotator, self).__init__()
+
+    def fit(self, X, Y=None, Z=None):
```

Comment:
Based on @Lovkush-A comment, I suggest:
```suggestion
    def fit(self, Z, Y=None, X=None):
```

Reply:
I disagree - I think this should be close to:

* the supervised setting as in `scikit-learn` - we use `fit` and `predict`
* the endogeneous/exogeneous labelling, where information also flows `X`->`Y`

Reply:
also, I think you annotated the wrong annotator? This is the panel annotator, which moved to #1036

Reply:
based on the suggestion in #1023, I introduced a `transform` method which asliases `predict`.

The `transform` method is compliant with the interface in our transformers, while `predict` mirrors forecasting and `scikit-learn`.